.github/workflows/tests.yaml

```diff
@@ -0,0 +1,55 @@
+name: Tests Pipeline
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/tests.yml'
+      - 'mcp_hydrolix/**'
+      - 'mcp_hydrolix/tests/**'
+      - 'mcp_hydrolix/pyproject.toml'
+      - 'mcp_hydrolix/ruff.toml'
+      - 'mcp_hydrolix/uv.lock'
+      - 'mcp_hydrolix/Dockerfile'
+      - 'mcp_hydrolix/docker-compose.yaml'
+  workflow_dispatch:
+
+jobs:
+  core-test:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/astral-sh/uv:0.8.9-python3.13-bookworm-slim
+    services:
+      clickhouse:
+    steps:
+      - uses: actions/checkout@v4
+      - name: Cache UV dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/uv
+          key: ${{ runner.os }}-uv-${{ hashFiles('mcp_hydrolix/uv.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-uv-
+      - name: Build and run Docker Compose services
+        run: |
+          apt-get update && apt-get install -y ca-certificates curl gnupg
+          install -m 0755 -d /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+          chmod a+r /etc/apt/keyrings/docker.gpg
+          echo \
+          "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+          $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+          tee /etc/apt/sources.list.d/docker.list > /dev/null
+          apt-get update
+          apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+          docker compose up -d --wait --wait-timeout 300
+      - name: Run tests
+        working-directory: .
+        run: |
+          uv sync --frozen --no-editable
+          # uvx ruff check --fix --exit-non-zero-on-fix --config ruff.toml .
+          uvx ruff format --check --diff --config ruff.toml .
+          uv pip install '.[dev]'
+          uv run pytest
+      - name: Stop Docker Compose services
+        if: always()
+        run: docker compose down
```

Comment:
Make sure to install and run `pre-commit`: you can catch up on the commits that are already pushed with `pre-commit run --all-files`